functions/endChat.ts

```diff
@@ -77,51 +77,53 @@ const updateTaskAssignmentStatus = async (
   context: Context<EnvVars>,
   event: Body,
 ) => {
-  const client = context.getTwilioClient();
-  let channelCleanupRequired = false;
-
-  // Fetch the Task to 'cancel' or 'wrapup'
-  const task = await client.taskrouter
-    .workspaces(context.TWILIO_WORKSPACE_SID)
-    .tasks(taskSid)
-    .fetch();
-
-  // Send a Message indicating user left the conversation
-  if (task.assignmentStatus === 'assigned') {
-    const endChatMessage = getEndChatMessage(event);
-    await context
-      .getTwilioClient()
-      .chat.services(context.CHAT_SERVICE_SID)
-      .channels(channelSid)
-      .messages.create({
-        body: endChatMessage,
-        from: 'Bot',
-        xTwilioWebhookEnabled: 'true',
-      });
-  }
-
-  // Update the task assignmentStatus
-  const updateAssignmentStatus = (assignmentStatus: TaskInstance['assignmentStatus']) =>
-    client.taskrouter
+  try {
```

Comment:
I wonder if it might make more sense to just return the updated task from this function and let the caller figure out what the follow on action is based on it's state, rather than this function telling the caller what to do next, which seems to be outside its responsibility?